pcsx2/Recording/InputRecording.cpp

```diff
@@ -0,0 +1,205 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2019  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+
+#include "AppSaveStates.h"
+#include "AppGameDatabase.h"
+#include "Common.h"
+#include "Counters.h"
+#include "MemoryTypes.h"
+#include "SaveState.h"
+
+#include "Recording/RecordingControls.h"
+#include "InputRecording.h"
+
+#include <vector>
+
+
+InputRecording g_InputRecording;
+
+// Tag and save framecount along with savestate
+void SaveStateBase::InputRecordingFreeze()
+{
+	FreezeTag("InputRecording");
+	Freeze(g_FrameCount);
+	if (IsLoading()) {
+		g_InputRecordingData.addUndoCount();
+	}
+}
+
+// Main func for handling controller input data
+// - Called by Sio.cpp::sioWriteController
+void InputRecording::ControllerInterrupt(u8 &data, u8 &port, u16 & bufCount, u8 buf[])
+{
+	// TODO - Multi-Tap Support
+	// Only examine controllers 1 / 2
+	if (port < 0 || 1 < port)
+		return;
+
+	//==========================
+	// This appears to try to ensure that we are only paying attention
+	// to the frames that matter, the ones that are reading from
+	// the controller.
+	//
+	// See - Lilypad.cpp:1254
```

Comment:
This comment will become obsolete if the code is updated with new lines. Maybe a function comment ..etc will be more fitting ?